source/Preferences.cpp

```diff
@@ -347,14 +349,18 @@ void Preferences::SetScrollSpeed(int speed)
 // View zoom.
 double Preferences::ViewZoom()
 {
-	return ZOOMS[zoomIndex];
+	const auto &zooms = GameData::Interfaces().Get("main view")->GetList("zooms");
+	if(zoomIndex >= zooms.size())
+		return zooms.empty() ? 1. : zooms.back();
+	return zooms[zoomIndex];
```

Comment:
Why not just get the list returned by Preferences::Zooms? This is a list we can guarantee isn't empty.
```suggestion
	const auto &zooms = Zooms();
	if(zoomIndex >= zooms.size())
		return zooms.back();
	return zooms[zoomIndex];
```